pebble_manager.py

```diff
@@ -36,6 +38,13 @@ def register_endpoints(self):
         self.pebble.register_endpoint("APPLICATION_LIFECYCLE", self.handle_lifecycle, preprocess=False)
         self.pebble.register_endpoint("LAUNCHER", self.handle_launcher, preprocess=False)
 
+    def determine_version_info(self):
+        self.watch_version_info = self.pebble.get_versions()
+        
+        version_str = self.watch_version_info['normal_fw']['version'][1:]
+        pieces = re.split("[\.\-]", version_str)
```

Comment:
The backslashes are unnecessary; special characters aren't special inside `[]`.

You should also prefix regexes with `r` — this creates a "raw string", in which backslash escapes are ignored. Otherwise you have to do `\\.`:

```python
r"[.-]"
```